packages/mui-base/src/useMenuItem/useMenuItem.ts

```diff
@@ -11,6 +11,10 @@ import {
 import { useListItem } from '../useList';
 import { useCompoundItem } from '../utils/useCompoundItem';
 
+function idGenerator(existingKeys: Set<string>) {
```

Comment:
Not really sure how to test that this function is being called or not as this function runs only on react versions less than 18.